bootstrap/eks/api/v1beta2/eksconfig_types.go

```diff
@@ -87,6 +110,207 @@ type EKSConfigStatus struct {
+	// +optional
+	PrimaryGroup *string `json:"primaryGroup,omitempty"`
+
+	// LockPassword specifies if password login should be disabled
+	// +optional
+	LockPassword *bool `json:"lockPassword,omitempty"`
+
+	// Sudo specifies a sudo role for the user
+	// +optional
+	Sudo *string `json:"sudo,omitempty"`
+
+	// SSHAuthorizedKeys specifies a list of ssh authorized keys for the user
+	// +optional
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+}
+
+// NTP defines input for generated ntp in cloud-init.
+type NTP struct {
+	// Servers specifies which NTP servers to use
+	// +optional
+	Servers []string `json:"servers,omitempty"`
+
+	// Enabled specifies whether NTP should be enabled
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+}
+
+// DiskSetup defines input for generated disk_setup and fs_setup in cloud-init.
+type DiskSetup struct {
+	// Partitions specifies the list of the partitions to setup.
+	// +optional
+	Partitions []Partition `json:"partitions,omitempty"`
+
+	// Filesystems specifies the list of file systems to setup.
+	// +optional
+	Filesystems []Filesystem `json:"filesystems,omitempty"`
+}
+
+// Partition defines how to create and layout a partition.
+type Partition struct {
+	// Device is the name of the device.
+	Device string `json:"device"`
+	// Layout specifies the device layout.
+	// If it is true, a single partition will be created for the entire device.
+	// When layout is false, it means don't partition or ignore existing partitioning.
+	Layout bool `json:"layout"`
+	// Overwrite describes whether to skip checks and create the partition if a partition or filesystem is found on the device.
+	// Use with caution. Default is 'false'.
+	// +optional
+	Overwrite *bool `json:"overwrite,omitempty"`
+	// TableType specifies the tupe of partition table. The following are supported:
+	// 'mbr': default and setups a MS-DOS partition table
+	// 'gpt': setups a GPT partition table
+	// +optional
+	TableType *string `json:"tableType,omitempty"`
+}
+
+// Filesystem defines the file systems to be created.
+type Filesystem struct {
+	// Device specifies the device name
+	Device string `json:"device"`
+	// Filesystem specifies the file system type.
+	Filesystem string `json:"filesystem"`
+	// Label specifies the file system label to be used. If set to None, no label is used.
+	Label string `json:"label"`
+	// Partition specifies the partition to use. The valid options are: "auto|any", "auto", "any", "none", and <NUM>, where NUM is the actual partition number.
+	// +optional
+	Partition *string `json:"partition,omitempty"`
+	// Overwrite defines whether or not to overwrite any existing filesystem.
+	// If true, any pre-existing file system will be destroyed. Use with Caution.
+	// +optional
+	Overwrite *bool `json:"overwrite,omitempty"`
+	// ReplaceFS is a special directive, used for Microsoft Azure that instructs cloud-init to replace a file system of <FS_TYPE>.
```

Comment:
I don't think this is needed here, right? :) We are specifically an AWS provider.